File: AuctionServices/src/main/java/com/artauction/auctionservice/Services/Interfaces/PostBidServices.java
package com.artauction.auctionservice.Services.Interfaces;

import com.artauction.auctionservice.Dtos.PostBid.BuyerPostBidDto;
import com.artauction.auctionservice.Dtos.PostBid.PostBidCreationDto;
import com.artauction.auctionservice.Dtos.PostBid.PostBidDto;
import com.artauction.auctionservice.Entities.PostSold;

import java.util.Collection;
import java.util.List;
import java.util.Optional;

public interface PostBidServices {
    List<PostBidDto> getAllPostBids(int artworkPostId);
    Collection<BuyerPostBidDto> getAllBuyerBids(int buyerId);
    boolean createPostBid(PostBidCreationDto postBidCreationDto);
    boolean deletePostBid(int artworkPostId, int buyerId);
}

File: Project/Backend/src/ArtAuction.Application/Services/PostBidServices.cs
using ArtAuction.Application.DTOs.ArtworkPost;
using ArtAuction.Application.Entities;
using ArtAuction.Application.Interfaces.Repositories;
using ArtAuction.Application.Interfaces.Services;
using AutoMapper;

namespace ArtAuction.Application.Services;

public class PostBidServices : IPostBidServices
{
    // Attributes
    private readonly IMapper _mapper;
    private readonly IPostBidRepo _postBidRepo;
    
    // Constructor
    public PostBidServices(IMapper mapper, IPostBidRepo postBidRepo)
    {
        this._mapper = mapper;
        this._postBidRepo = postBidRepo;
    }
    
    // Methods
    public async Task<List<PostBidDto>> GetAllPostBids(int artworkPostId)
    {
        // Get post bids
        var postBids = await _postBidRepo.GetAllPostBids(artworkPostId);
        
        // Return mapped DTO
        return _mapper.Map<List<PostBidDto>>(postBids);
    }

    public async Task<ICollection<BuyerPostBidDto>> GetAllBuyerBids(int buyerId)
    {
        // Get post bids
        var postBids = await _postBidRepo.GetAllBuyerBids(buyerId);
        
        // Return mapped DTO
        return _mapper.Map<ICollection<BuyerPostBidDto>>(postBids);
    }

    public async Task<bool> CreatePostBid(PostBidCreationDto postBidCreationDto)
    {
        // Map DTO to entity
        var postBid = _mapper.Map<PostBid>(postBidCreationDto);
        
        // Return boolean
        return await _postBidRepo.CreatePostBid(postBid);
    }

    public async Task<bool> DeletePostBid(int artworkPostId, int buyerId)
    {
        // Delete entity and return boolean
        return await _postBidRepo.DeletePostBid(artworkPostId, buyerId);
    }
}